### PR TITLE
Add packagegen tool tests

### PR DESCRIPTION
**Pull request by AnjanaSenanayake**
Add packagegen tool tests

### CODE REVIEW

File: native/fhir-to-bal-lib/src/test/resources/ballerina.tests/test_constraints.bal [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+        }, 
+        participant: []
+    };
+    USCoreCareTeam|constraint:Error validate = constraint:validate(careTeam, USCoreCareTeam);
+    if validate is USCoreCareTeam {
+        test:assertFail("careTeam should not be valid");
+    } else {
+        test:assertTrue(true, "careTeam is not valid");
+    }
+}
```
Comment (isuruh15) [inline native/fhir-to-bal-lib/src/test/resources/ballerina.tests/test_constraints.bal:18]: missing EoF

File: native/fhir-to-bal-lib/src/test/resources/ballerina.tests/test_extended_profiles.bal [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            r4:Coding coding = codings[0];
+            test:assertEquals(coding.system, "http://terminology.hl7.org/CodeSystem/observation-category");
+            test:assertEquals(coding.code, "vital-signs");
+        } else {
+            test:assertFail("codings is not populated with fixed values");
+        }
+    } else {
+        test:assertFail("category is not populated with fixed values");
+    }
+}
```
Comment (isuruh15) [inline native/fhir-to-bal-lib/src/test/resources/ballerina.tests/test_extended_profiles.bal:71]: check other places as well

File: native/health-cli/pom.xml [hunk 3]
```diff
@@ -99,6 +99,90 @@
...
+                        <goals>
+                            <goal>copy</goal>
+                            <goal>unpack</goal>
+                        </goals>
+                        <configuration>
+                            <artifactItems>
+                                <artifactItem>
+                                    <groupId>io.ballerina</groupId>
+                                    <artifactId>fhir-to-bal-lib</artifactId>
+                                    <version>1.0.1-SNAPSHOT</version>
```
Comment (sameeragunarathne) [inline native/health-cli/pom.xml:118]: Let's use the project version?

File: native/fhir-to-bal-lib/src/test/resources/ballerina.tests/test_constraints.bal (hunk 1 above)
Comment (AnjanaSenanayake) [reply native/fhir-to-bal-lib/src/test/resources/ballerina.tests/test_constraints.bal]: Fixed in https://github.com/ballerina-platform/fhir-tools/commit/653cd017f886ca3722c492aabb4fa00b95c38184

File: native/fhir-to-bal-lib/src/test/resources/ballerina.tests/test_extended_profiles.bal (hunk 2 above)
Comment (AnjanaSenanayake) [reply native/fhir-to-bal-lib/src/test/resources/ballerina.tests/test_extended_profiles.bal]: Fixed in https://github.com/ballerina-platform/fhir-tools/commit/653cd017f886ca3722c492aabb4fa00b95c38184

File: native/health-cli/pom.xml (hunk 3 above)
Comment (AnjanaSenanayake) [reply native/health-cli/pom.xml]: Fixed in https://github.com/ballerina-platform/fhir-tools/commit/653cd017f886ca3722c492aabb4fa00b95c38184

**Review by sameeragunarathne — APPROVED** [review]

File: native/health-cli/pom.xml [hunk 4]
```diff
@@ -99,7 +99,226 @@
                     </archive>
                 </configuration>
             </plugin>
+            <plugin>
```
Comment (ThishaniLucas) [inline native/health-cli/pom.xml:102]: 102-184 Do we need these outside the profile? We should be able to generate the package and run the tests only when the test profile is active.

Comment (AnjanaSenanayake) [reply native/health-cli/pom.xml]: Mistake. Fixed in https://github.com/ballerina-platform/fhir-tools/commit/d4aa6b814a9d9dbabb8c56bc4730d91e8afd1906

**Review by sameeragunarathne — APPROVED** [review]

**Review by sameeragunarathne — APPROVED** [review]

**Pull request merged**